src/Sarif/ExtensionMethods.cs

```diff
@@ -278,18 +281,42 @@ public static string GetFirstSentence(string text)
                         {
                             withinParentheses = false;
                         }
+                        lastEncounteredWasDot = false;
                         break;
                     }
+
+                    case '.':
+                    {
+                        if (withinQuotes || withinParentheses) { continue; }
+                        lastEncounteredWasDot = true;
+                        break;
+                    }
+
+                    // If we encounter a line-break, we return all leading text.
                     case '\n':
                     case '\r':
-                    case '.':
                     {
                         if (withinQuotes || withinParentheses) { continue; }
-                        return text.Substring(0, length).TrimEnd('\r', '\n');
+                        return text.Substring(0, length).TrimEnd('\r', '\n', ' ', '.') + ".";
```

Comment:
Consider extracting a method `EnsureEndingPeriod` since this calculation occurs twice.